tests/Oryx.Tests.Common/DockerCli.cs

```diff
@@ -147,6 +150,25 @@ public DockerRunCommandProcessResult RunAndDoNotWaitForProcessExit(DockerRunArgu
                 $"{DockerCmd} {string.Join(" ", arguments)}");
         }
 
+        public async Task<DockerRunCommandProcessResult> RunAndWaitForContainerStartAsync(DockerRunArguments dockerRunArguments)
+        {
+            var dockerCommand = RunAndDoNotWaitForProcessExit(dockerRunArguments);
+
+            var retry = 0;
+            while (!dockerCommand.Process.HasExited && retry < TotalContainerStartupRetries)
+            {
+                if (!string.IsNullOrEmpty(GetContainerStatus(dockerCommand.ContainerName)))
```

Comment:
Trying to have better understanding for this part, It'd be great if you could help clarifying a bit what does this part do.

Reply:
I added a comment to explain why we are getting the container status, but essentially the entire method is checking every few seconds for the existence of the container, and as soon as it exists (or the process exits, indicating a failure) then we return the container result. This method allows us to go into the GetHostPortAsyc method, knowing that either the container exists and is running or something has errored. I tried to explain this flow in this comment https://github.com/microsoft/Oryx/pull/1471#discussion_r917934755